src/ecs/resources/input.rs

```diff
@@ -333,9 +333,9 @@ impl InputHandler {
     }
 
     /// Returns the value of an axis by the i32 id, if the id doesn't exist this returns None.
-    pub fn axis_value(&self, id: i32) -> Option<f32> {
+    pub fn axis_value(&self, id: &str) -> Option<f32> {
```

Comment:
I disagree. It's more difficult to read and you'll almost always pass a literal. If we argue with "it can save the user a method call sometimes", then Rust's choice being explicit about conversions doesn't make sense.